boot/freeldr/freeldr/lib/fs/iso.c

```diff
@@ -529,6 +524,16 @@ const DEVVTBL* IsoMount(ULONG DeviceId)
         return NULL;
     }
 
+    Volume = FrLdrTempAlloc(sizeof(*Volume), TAG_ISO_VOLUME);
+    if (!Volume)
+        return NULL;
+    RtlZeroMemory(Volume, sizeof(*Volume));
+
+    IsoVolumes[DeviceId] = Volume;
+
+    Volume->PvdDirectorySector = Pvd->RootDirRecord.ExtentLocationL;
+    Volume->PvdDirectoryLength = Pvd->RootDirRecord.DataLengthL;
```

Comment:
```suggestion
    /* Cache the volume info */
    Volume = FrLdrTempAlloc(sizeof(*Volume), TAG_ISO_VOLUME);
    if (!Volume)
        return NULL;
    RtlZeroMemory(Volume, sizeof(*Volume));
    Volume->PvdDirectorySector = Pvd->RootDirRecord.ExtentLocationL;
    Volume->PvdDirectoryLength = Pvd->RootDirRecord.DataLengthL;
    IsoVolumes[DeviceId] = Volume;
```

Reply:
(Note: Serge's suggestion is in order to have the pointer inserted into the `IsoVolumes` array once it's fully initialized. A bit like what's done in the other FSes.)

Reply:
Fixed.